src/Report/JsonReport.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Report;
+
+/**
+ * @author Boris Gorbylev <ekho@ekho.name>
+ *
+ * @internal
+ */
+final class JsonReport implements ReportInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getFormat()
+    {
+        return 'json';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate(ReportConfig $reportConfig)
+    {
+        $jFiles = array();
+
+        foreach ($reportConfig->getChanged() as $file => $fixResult) {
+            $jfile = array('name' => $file);
+
+            if ($reportConfig->shouldAddAppliedFixers()) {
+                $jfile['appliedFixers'] = $fixResult['appliedFixers'];
+            }
+
+            if (!empty($fixResult['diff'])) {
+                $jfile['diff'] = $fixResult['diff'];
+            }
+
+            $jFiles[] = $jfile;
+        }
+
+        $json = array(
+            'files' => $jFiles,
+        );
+
+        if (null !== $reportConfig->getTime()) {
+            $json['time'] = array(
+                'total' => round($reportConfig->getTime() / 1000, 3),
```

Comment:
It's not about configuration it's a result, so what about `ReportSummary`?

And the `ReportInterface` indicates it holds information while it actually reports (like a reporter or ReportFormatter).

--
When class performs an action (rather then keeping data) it should end `-er` or `-or`,
Controller, Reporter, Configurator, Fixer. Factory being in exception or course ;)

Credit: https://youtu.com/--L0bS8n5ow?t=26m38s (highly recommended video)